### ISSUE TITLE
License

### ISSUE THREAD

**Issue by streamich** [josephg/crdt-benchmarks#5]
It is not clear under what license this repo can be used. A number of traces mention [CC-4.0](https://creativecommons.org/licenses/by/4.0/), but under other traces there is no mention of a license.

Would it be possible to clarify the license for the entire repo?

**Comment by josephg** [comment]
Yeah I'd like everything in this repository to be CC-4 or something similar. I've personally made all of these traces except for the automerge-perf trace. And I've just added the license tag to the README of the other traces that were missing it. - Except for automerge-perf, which I didn't make so I can't license it.

The automerge-perf trace comes from [automerge/automerge-perf](https://github.com/automerge/automerge-perf/) by @ept. The source text was a paper that is now published, so there's probably some complex copyright around that.

Maybe Martin can chime in - are you happy to put your automerge-perf editing trace under a public license of some sort?

**Comment by ept** [comment]
Agree it would be great to license everything CC. I just checked and we did assign the copyright for that article to the [IEEE](https://www.ieee.org/). The relevant part of the copyright assignment reads:

> Authors/employers may reproduce or authorize others to reproduce the Work, material extracted verbatim from the Work, or derivative works for the author's personal use or for company use, provided that the source and the IEEE copyright notice are indicated, the copies are not used in any way that implies IEEE endorsement of a product or service of any employer, and the copies themselves are not offered for sale. Although authors are permitted to re-use all or portions of the Work in other works, this does not include granting third-party requests for reprinting, republishing, or other types of re-use.

So on the one hand I'm allowed to "authorize others to reproduce the Work", but not "grant third-party requests for re-use". Not sure how to interpret that.

We can probably treat the editing trace the same way as a preprint PDF, for which the copyright assignment says:

> Before submitting an article to an IEEE publication, authors frequently post their manuscripts to their own web site, their employer's site, or to another server that invites constructive comment from colleagues. Upon submission of an article to IEEE, an author is required to transfer copyright in the article to IEEE, and the author must update any previously posted version of the article with a prominently displayed IEEE copyright notice. Upon publication of an article by the IEEE, the author must replace any previously posted electronic versions of the article with either (1) the full citation to the IEEE work with a Digital Object Identifier (DOI) or link to the article abstract in IEEE Xplore, or (2) the accepted version only (not the IEEE-published version), including the IEEE copyright notice and full citation, with a link to the final, published article in IEEE Xplore.

I think a reasonable interpretation would be that we can license the editing trace [CC BY-NC](https://creativecommons.org/licenses/by-nc/4.0/), which would allow the data to be distributed but not sold. As long as we also say "(c) IEEE" somewhere suitable and link to the [official publication page](https://ieeexplore.ieee.org/document/7909007/), it should be fine.

**Comment by josephg** [comment]
Great - I’ll add a comment to that effect. Thanks @ept

**Comment by mohe2015** [comment]
@josephg It seems like this license statement was never added. @ept has now licensed the editing trace under [CC BY](https://creativecommons.org/licenses/by/4.0/) (see https://github.com/automerge/automerge-perf), so this should now be legally possible. Would you like to add that license statement please? Thanks.

**Comment by ept** [comment]
On further thought I think we can license it CC-BY. No need to make it CC-BY-NC, since I believe the copyright assignment concerns only the final paper, and the editing trace contains a lot more information than the final paper. I've now made this explicit in the automerge-perf repo.
